### PR TITLE
Fix copy/pasted modules

### PR DESCRIPTION
**Pull request by oOKexOo**
**When merged this pull request will:**
- Fix activation of copy/pasted modules (_e.g._ smoke shell)

### CODE REVIEW

File: addons/editor/functions/fnc_bi_curatorObjectPlaced.sqf [hunk 1]
```diff
@@ -28,6 +29,11 @@ if (GVAR(unitRadioMessages) == 0 && {!isNull _group && {side _group in [west, ea
     [effectiveCommander _object, "CuratorObjectPlaced"] call BIS_fnc_curatorSayMessage;
 };
 
+if (_object isKindOf "Module_f") then
+{
```
Comment (jonpas) [inline addons/editor/functions/fnc_bi_curatorObjectPlaced.sqf]: On same line.

File: addons/editor/functions/fnc_bi_curatorObjectPlaced.sqf [hunk 2]
```diff
@@ -28,6 +29,11 @@ if (GVAR(unitRadioMessages) == 0 && {!isNull _group && {side _group in [west, ea
...
+	_object setVariable ["BIS_fnc_initModules_activate", true, true];
```
Comment (jonpas) [inline addons/editor/functions/fnc_bi_curatorObjectPlaced.sqf]: Wrong indent.

File: addons/editor/functions/fnc_bi_curatorObjectPlaced.sqf [hunk 3]
```diff
@@ -28,6 +29,10 @@ if (GVAR(unitRadioMessages) == 0 && {!isNull _group && {side _group in [west, ea
     [effectiveCommander _object, "CuratorObjectPlaced"] call BIS_fnc_curatorSayMessage;
 };
 
+if (_object isKindOf "Module_f") then {
```
Comment (mharis001) [inline addons/editor/functions/fnc_bi_curatorObjectPlaced.sqf]:
```suggestion
if (_object isKindOf "Module_F") then {
```

**Comment by mharis001** [comment]
Since ZEN modules do not rely on BI's module framework but still inherit from `Module_F`, I think this should also check: `!(_object isKindOf QGVAR(moduleBase)` to reduce the number variable broadcasts needed. And maybe move this to the modules component?

**Comment by oOKexOo** [comment]
Ok, I could add a new class EH to the curator module for attaching a `CuratorObjectPlaced` EH, which in return calls something like `fnc_curatorModulePlaced.sqf`. All located in module component ofc.

File: addons/modules/XEH_preInit.sqf [hunk 4]
```diff
@@ -6,4 +6,15 @@ PREP_RECOMPILE_START;
 #include "XEH_PREP.hpp"
 PREP_RECOMPILE_END;
 
+["ModuleCurator_F", "Init", {
+    params ["_logic"];
+
+    _logic addEventHandler ["CuratorObjectPlaced",
+    {
```
Comment (mharis001) [inline addons/modules/XEH_preInit.sqf]: Same line.

File: addons/modules/XEH_preInit.sqf [hunk 5]
```diff
@@ -6,4 +6,15 @@ PREP_RECOMPILE_START;
...
+        if (param[1] isKindOf "Module_F") then {
```
Comment (mharis001) [inline addons/modules/XEH_preInit.sqf]: Use params. `params ["", "_object"];`

File: addons/modules/XEH_preInit.sqf [hunk 6]
```diff
@@ -6,4 +6,15 @@ PREP_RECOMPILE_START;
...
+            _this call FUNC(handleModulePlaced);
```
Comment (mharis001) [inline addons/modules/XEH_preInit.sqf]: Don't think we need a separate function for this. Can just do the `&& {!(_logic isKindOf QGVAR(moduleBase))}` check in the if statement above.

File: addons/modules/XEH_preInit.sqf [hunk 7]
```diff
@@ -6,4 +6,15 @@ PREP_RECOMPILE_START;
 #include "XEH_PREP.hpp"
 PREP_RECOMPILE_END;
 
+["ModuleCurator_F", "Init", {
```
Comment (mharis001) [inline addons/modules/XEH_preInit.sqf]: Comment above this line maybe?
`// Fix copy/pasted modules that use BI's module framework`

File: addons/modules/XEH_preInit.sqf (hunk 4 above)
Comment (oOKexOo) [reply addons/modules/XEH_preInit.sqf]: Damn, still need to get used to K&R style.

**Pull request merged**